### PR TITLE
docs: explain Renovate config for GitHub Merge Queue

### LINKED ISSUE #20321: Explain how to get GitHub's pull request merge queue working with Renovate
### What would you like Renovate to be able to do?

Copy/pasting @rarkins mission statement: [^statement]

> Something that means multiple PRs don't need to be rebased after each merge.

[^statement]: https://github.com/renovatebot/renovate/issues/6011#issue-603228248

#### Partially working already [^quote]

> Actually, Renovate can already add to GitHub merge queues. It is necessary to enable auto merge in GitHub's repo settings (even though a merge queue is being used). Renovate is already making the right API call:
> 
> https://github.com/renovatebot/renovate/blob/bde951257a79737351859d1c08db0ad214e61494/lib/modules/platform/github/index.ts#L1455-L1458
> 
> So it ends up looking like so:
> 
> github renovate
> 
> This is using platform auto merge:
> 
> ```json
> {
>     "automerge": true,
>     "automergeType": "pr",
>     "platformAutomerge": true
> }
> ```

#### Who can use Merge Queues?

> Pull request merge queues are available in any public repository owned by an organization, or in private repositories owned by organizations using GitHub Enterprise Cloud. [^docs]

[^docs]: https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue
[^quote]: https://github.com/renovatebot/renovate/issues/6011#issuecomment-1424561440

#### Relevant documentation:

- [GitHub Docs, Merge Queue](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue)
- [GitHub Docs, `merge_group` webhook/event](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#merge_group)
- [GitHub Docs, `merge_group` explanation for end-users](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#merge_group)
- [GitHub Blog about `merge_group`](https://github.blog/changelog/2022-08-18-merge-group-webhook-event-and-github-actions-workflow-trigger/)

#### Todos

- [x] Explain current support for GitHub Merge Queue in our docs
- [x] Discover what works/doesn't work right now
- [x] Make todo list to get the feature working fully

### Is this a feature you are interested in implementing yourself?

No

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Add new sections in the Key Concepts, Automerge page to explain how to get Renovate working with GitHub's Merge Queue
- Link to the new sections from the `platformAutomerge` docs

## Context

Closes #20321.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — DISMISSED** [review]
like it, we probably should add a note that it's a beta feature?

File: docs/usage/configuration-options.md [hunk 1]
```diff
@@ -2147,6 +2147,8 @@ For example, GitHub might automerge a Renovate branch even if it's behind the ba
 
 Please check platform specific docs for version requirements.
 
+To learn how to use GitHub's Merge Queue feature with Renovate, read the [Key Concepts page, GitHub Merge Queue](./key-concepts/automerge.md#github-merge-queue) page.
```
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:2150]:
```suggestion
To learn how to use GitHub's Merge Queue feature with Renovate, read our [Key Concepts, Automerge, GitHub Merge Queue](./key-concepts/automerge.md#github-merge-queue) docs.
```

**Comment by HonkingGoose** [comment]
I've added a note that says GitHub's Merge Queue is labeled as beta by GitHub. 😉

**Review by viceice — DISMISSED** [review]

**Comment by vchernin** [comment]
For me this feature didn't work unless [Auto Merge was allowed in repository settings](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-auto-merge-for-pull-requests-in-your-repository#managing-auto-merge). Maybe that is worth mentioning in the docs too?

File: docs/usage/key-concepts/automerge.md [hunk 2]
```diff
@@ -103,6 +103,66 @@ For example:
...
+1. On `github.com`:
+   1. Go to your repository's branch protection rules for your base branch (usually `main`) and enable the "Require merge queue" setting
+   1. Confirm you've set the correct "required checks" for your base branch
+1. Allow Renovate to automerge by setting `automerge=true` and `platformAutomerge=true` in your Renovate config file, for example:
+
+   ```json
+   {
+     "automerge": true,
+     "platformAutomerge": true
+   }
```
Comment (rarkins) [inline docs/usage/key-concepts/automerge.md:146]:
```suggestion
   {
     "platformAutomerge": true,
     "packageRules": [{
       "description": "Automerge non-major updates",
       "matchUpdateTypes": ["minor", "patch"],
       "automerge": true
     }]
   }
```

Recommending automerge=true at root level is not a good idea!

File: docs/usage/key-concepts/automerge.md [hunk 3]
```diff
@@ -103,6 +103,66 @@ For example:
...
+   ```
```
Comment (rarkins) [inline docs/usage/key-concepts/automerge.md:164]:
```suggestion
1. Allow Renovate to automerge by setting `automerge=true` and `platformAutomerge=true` in your Renovate config file (see earlier example)
   ```
```

**Review by rarkins — CHANGES_REQUESTED** [review]

**Comment by HonkingGoose** [comment]
> **build / lint (pull_request)** Cancelled after 14m 

It took nearly 15 minutes for the _Set up Node.js 18_ step, and then the whole thing timed-out. Looks like there's a problem with downloading the Node.js cached files. At first the download is fast, but it drops to 0.1 mb/s quickly. Logs: https://github.com/renovatebot/renovate/actions/runs/4203138015/jobs/7292162459

**Review by rarkins — APPROVED** [review]

**Comment by HonkingGoose** [comment]
I think we're good to go now. 😉

**Pull request merged**